bundles/storage/org.eclipse.smarthome.storage.json/src/main/java/org/eclipse/smarthome/storage/json/JsonStorage.java

```diff
@@ -186,8 +192,7 @@ private T deserialize(Map<String, Object> jsonValue) {
                 loadedValueType = (Class<T>) classLoader.loadClass((String) jsonValue.get(CLASS));
             }
 
-            String jsonString = mapper.toJson(jsonValue.get("value"));
-            value = mapper.fromJson(jsonString, loadedValueType);
+            value = innerMapper.fromJson((JsonElement) jsonValue.get(VALUE), loadedValueType);
```

Comment:
yes. Which makes me asking myself why the internal data structure carrying class & value has to be a map at all. It would be much less confusing if it were a dedicated class. Let me quickly check what it will feel like...